File: Task/Fairshare-between-two-and-more/C-sharp/fairshare-between-two-and-more.cs
using System;
using System.Collections.Generic;

class FairshareBetweenTwoAndMore
{
    static void Main(string[] args)
    {
        foreach (int baseValue in new List<int> { 2, 3, 5, 11 })
        {
            Console.WriteLine($"Base {baseValue} = {string.Join(", ", ThueMorseSequence(25, baseValue))}");
        }
    }

    private static List<int> ThueMorseSequence(int terms, int baseValue)
    {
        List<int> sequence = new List<int>();
        for (int i = 0; i < terms; i++)
        {
            int sum = 0;
            int n = i;
            while (n > 0)
            {
                // Compute the digit sum
                sum += n % baseValue;
                n /= baseValue;
            }
            // Compute the digit sum modulo baseValue.
            sequence.Add(sum % baseValue);
        }
        return sequence;
    }
}

File: Task/Fairshare-between-two-and-more/PascalABC.NET/fairshare-between-two-and-more.pas
function ThueMorseSequence(terms, baseValue: integer): list<integer>;
begin
  result := new List<integer>;
  for var i := 0 to terms - 1 do
  begin
    var sum := 0;
    var n := i;
    while (n > 0) do
    begin
      // Compute the digit sum
      sum += n mod baseValue;
      n := n div baseValue;
    end;
    // Compute the digit sum modulo baseValue.
    result.Add(sum mod baseValue);
  end;
end;

begin
  foreach var baseValue in |2, 3, 5, 11| do
    println('Base', baseValue, '=', ThueMorseSequence(25, baseValue));
end.
